api/konnect/v1alpha1/konnect_cloudgateway_dataplane_configuration_types.go

```diff
@@ -0,0 +1,257 @@
+	// Name of the environment variable field to set for the data-plane group. Must be prefixed by KONG_.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern="^KONG_."
+	Name string `json:"name"`
+	// Value assigned to the environment variable field for the data-plane group.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	Value string `json:"value"`
+}
+
+// ConfigurationDataPlaneGroupAutoscale specifies the autoscale configuration for the data-plane group.
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleStatic' ? has(self.static) : true",message="static is required when type is ConfigurationDataPlaneGroupAutoscaleStatic"
+// +kubebuilder:validation:XValidation:rule="self.type == 'ConfigurationDataPlaneGroupAutoscaleAutopilot' ? has(self.autopilot) : true",message="autopilot is required when type is ConfigurationDataPlaneGroupAutoscaleAutopilot"
+type ConfigurationDataPlaneGroupAutoscale struct {
+	// ConfigurationDataPlaneGroupAutoscaleStatic specifies the static configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleStatic *ConfigurationDataPlaneGroupAutoscaleStatic `json:"static,omitempty"`
+
+	// ConfigurationDataPlaneGroupAutoscaleAutopilot specifies the autoscale configuration for the data-plane group.
+	//
+	// +kubebuilder:validation:Optional
+	ConfigurationDataPlaneGroupAutoscaleAutopilot *ConfigurationDataPlaneGroupAutoscaleAutopilot `json:"autopilot,omitempty"`
+
+	// Type of autoscaling to use.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=ConfigurationDataPlaneGroupAutoscaleStatic;ConfigurationDataPlaneGroupAutoscaleAutopilot
+	Type sdkkonnectcomp.ConfigurationDataPlaneGroupAutoscaleType `json:"type"`
+}
+
+// ConfigurationDataPlaneGroupAutoscaleAutopilot specifies the autoscale configuration for the data-plane group.
+type ConfigurationDataPlaneGroupAutoscaleAutopilot struct {
+	// Base number of requests per second that the deployment target should support.
+	//
+	// +kubebuilder:validation:Required
+	BaseRps int64 `json:"base_rps"`
+
+	// Max number of requests per second that the deployment target should support. If not set, this defaults to 10x base_rps.
+	MaxRps *int64 `json:"max_rps,omitempty"`
+}
+
+// ConfigurationDataPlaneGroupAutoscaleStatic specifies the static configuration for the data-plane group.
+type ConfigurationDataPlaneGroupAutoscaleStatic struct {
+	// Instance type name to indicate capacity.
+	//
+	// +kubebuilder:validation:Required
+	InstanceType sdkkonnectcomp.InstanceTypeName `json:"instance_type"`
+
+	// Number of data-planes the deployment target will contain.
+	//
+	// +kubebuilder:validation:Required
+	RequestedInstances int64 `json:"requested_instances"`
```

Comment:
maybe we can set a default value here as well, instead of requiring users to specify it. 1 instance as default would make sense I guess?

Reply:
Ditto: no default in the OAS for this field: https://github.com/Kong/sdk-konnect-go/blob/d511d983a17fb1be7ddd4950cc571f4482990127/openapi.yaml#L10602-L10605